### PR TITLE
Eliminate some redundant updates in SwiftUI components

### PR DESCRIPTION
**Pull request by tgoyne**
While adding Sendable annotations to the SwiftUI stuff I extracted some duplicated code and fixed some minor issues. Updating the search text for searchable triggered two updates because it modified both `filter` and `where`, which each triggered an update. The `where` property on the storage appears to be just totally redundant.

Changing anything on ObservableResults would trigger up to three updates due to storing temporary values in the `@StateObject` property.

AsyncOpen and AutoOpen would start an async open twice if both the partitionValue and configuration environment keys were set.

### CODE REVIEW

File: CHANGELOG.md
Comment (dianaafanador3) [inline CHANGELOG.md:34]: Should we link the PR for this three fixes.

File: RealmSwift/SwiftUI.swift
Comment (dianaafanador3) [inline RealmSwift/SwiftUI.swift:278]: I would probably add a message or comment so we know this is been override by the storage on each property wrappers

File: RealmSwift/SwiftUI.swift
Comment (dianaafanador3) [inline RealmSwift/SwiftUI.swift:477]: Nice catch on this

File: CHANGELOG.md
Comment (dianaafanador3) [inline CHANGELOG.md:38]: I would probably mention this fix also applies to @ObservedSectionedResults

File: RealmSwift/SwiftUI.swift
Comment (dianaafanador3) [inline RealmSwift/SwiftUI.swift:282]: Maybe it is too much work, but now that we are unifying storage for @ObservedResults and @ObservedSectionResults, there seems to be a request for extending @ObservedResults API to allow more than one sortDescriptor, https://github.com/realm/realm-swift/issues/7709,

**Review by dianaafanador3 — APPROVED** [review]
Nice PR, This would make our SwiftUI implementation more efficient. Just some small comments

File: RealmSwift/SwiftUI.swift
Comment (dianaafanador3) [inline RealmSwift/SwiftUI.swift:655]: Removing this and move it to `didSet` results in this function been called endlessly from `wrappedValue`. This also applies to `@ObservedSectionedResults`. I tested this on a project and restoring this and removing it from didSet solves the issue

**Review by dianaafanador3 — CHANGES_REQUESTED** [review]
I tested this PR with a project and the setupValue function is been called recurrently. This is prob

File: RealmSwift/SwiftUI.swift
Comment (tgoyne) [reply RealmSwift/SwiftUI.swift:655]: Good catch; I accidentally made it so that the only place `setupHasRun = true` was set was inside a `if setupHasRun { }` block. I don't see an obvious way to catch this from a test because I think it still always gave correct results and was just a lot slower than it should be.

File: RealmSwift/SwiftUI.swift
Comment (tgoyne) [reply RealmSwift/SwiftUI.swift:282]: It looks easy to implement but not so entirely trivial that I might as well just do it here.

Comment (dianaafanador3) [reply RealmSwift/SwiftUI.swift:289]: @tgoyne If you say you'll implement it here?, I'll wait for the final review

Comment (tgoyne) [reply RealmSwift/SwiftUI.swift:289]: No, I won't be implementing it here. I put a few too many conditionals in that sentence.

**Review by dianaafanador3 — APPROVED** [review]

**Pull request merged**
